### ISSUE TITLE
TODO Master issue

### ISSUE THREAD

**Issue by nipunbatra** [distillpub/post--bayesian-optimization#2]
- [x] Write about Hyperparameter v/s parameter: https://en.wikipedia.org/wiki/Hyperparameter_(machine_learning)
- [x] Write about Thompson sampling
- [x] Motivate grid search expensive computation cost
- [x] Formally introduce the properties/problem of Bayesian optimization (aka Frazier video) and relate each point to the gold mining problem. This should happen after we have done the gold mining problem and more generally decide to introduce the problem.
- [x] Explain role of epsilon in PI
- [x] epsilon in EI..
- [x] Write about galore of packages and services including sigopt, hyperopt, spearmint, scikit-optimize etc. 
- [x] Look into examples introduced in sigopt/hyperopt/spearmint/skoptimize to seek inspiration
- [x]  BO v/s GD: https://stats.stackexchange.com/questions/161923/bayesian-optimization-or-gradient-descent This also has some nice things which can be problematic for GPs (like being cubic in input dimension)
- [x] Use prettier fonts and experiment with five thirty eight style for figures
- [x] Refer and read this excellent article: https://thuijskens.github.io/2016/12/29/bayesian-optimisation/ and the citations in it.
- [x] Introduce the problems (kernel choosing, other hyperparameters) in Bayesian optimization. -- will refer slide# 30 in Peter Fraizer Bayesian Optimization tutorial, [link](https://people.orie.cornell.edu/pfrazier/Presentations/2018.11.INFORMS.tutorial.pdf)
- [x] Why the above problem is much easier to deal with?
- [x] Different Epsilon and the result on PI
- [x] BO when there are categorical variables also: https://github.com/scikit-optimize/scikit-optimize/issues/580 and hyperopt has an implementation, so does sigopt..
- [x] Neural network hyper-parameter search for simple CNN for MNIST: i) learning rate (0.001, 0.01, ...) - this would require us to probably use a log-scale to make the search linear; ii) batch size: 1, 2, 4, .. use a log scale base 2; iii) number of hidden units: maybe again log scale
- [x] Continuing from above - we can also make use of gradient information when available. This should be only a discussion, but should be good for us to know (after submission!)
- [x] Explain intuition of GP-UCB
- [x] Like SVM, create an optimisation on Random Forests with the two params being # estimators and max depth
- ~~[ ] Graph showing time saved.~~
